src/rbs_string.c

```diff
@@ -5,61 +5,37 @@
 #include <stdio.h>
 #include <ctype.h>
 
-rbs_string_t rbs_string_shared_new(const char *start, const char *end) {
+rbs_string_t rbs_string_new(const char *start, const char *end) {
     return (rbs_string_t) {
         .start = start,
         .end = end,
```

Comment:
So if I understand correctly, this is still a "shared"/"unowned" string, it just doesn't need to be tagged as such, because we're never going to `free()` any RBS strings anyway.

We might still want the shared/owned distinction for debugging, but this makes sense to me.

Reply:
Yes a string will be either "owned by the allocator" or "owned by something else" but we don't have to distinguish them anymore as the freeing is not manual anymore.

I prefer not to leave code around just for potential debugging purpose tho.